File: src/gestorAplicacion/administracion/Medicamento.java
/* Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
y Santiago Arboleda Acevedo */
package gestorAplicacion.administracion;

import gestorAplicacion.personas.Enfermedad;

import java.io.Serializable;

/* Clase usada en FormulaMedica, practicamente es el pilar de esta funcionalidad porque son los
medicamentos que se eligen allí */
public class Medicamento implements Serializable {
    // Atributos
    private String nombre;
    private Enfermedad enfermedad;
    private String descripcion;
    private int cantidad;
    private float precio;

    // Constructor
    public Medicamento (String nombre, Enfermedad enfermedad, String descripcion, int cantidad, float precio) {
        this.nombre = nombre;
        this.enfermedad = enfermedad;
        this.descripcion = descripcion;
        this.cantidad = cantidad;
        this.precio = precio;
    }

    // Métodos

    //Método que elimina una unidad de la cantidad total de medicamentos
    public void eliminarCantidad() {
        this.cantidad -= 1;
    }

    // Getter y setter
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public Enfermedad getEnfermedad() {
        return enfermedad;
    }

    public void setEnfermedad(Enfermedad enfermedad) {
        this.enfermedad = enfermedad;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public int getCantidad() {
        return cantidad;
    }

    public void setCantidad(int cantidad) {
        this.cantidad = cantidad;
    }

    public float getPrecio() {
        return precio;
    }

    public void setPrecio(float precio) {
        this.precio = precio;
    }
    // toString
    @Override
    public String toString() {
        return "Nombre: " + nombre + ", " + "Enfermedad: " + enfermedad.getNombre() + " " + enfermedad.getTipologia() + ", " + "Descripcion: " + descripcion;
    }
}

File: Python/src/gestor_aplicacion/administracion/medicamento.py
# Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
# y Santiago Arboleda Acevedo

# Clase usada en FormulaMedica, practicamente es el pilar de esta funcionalidad porque son los
# medicamentos que se eligen allí
class Medicamento:
    # Atributos y constructor
    def __init__(self, nombre, enfermedad, descripcion, cantidad, precio):
        self._nombre = nombre
        self._enfermedad = enfermedad
        self._descripcion = descripcion
        self._cantidad = cantidad
        self._precio = precio
    # Metodos

    # Elimina una unidad la cantidad del medicamento
    def eliminar_cantidad(self):
        self._cantidad -= 1

    # Muestra la informacion del medicamento sin la cantidad
    def mostrar_info(self):
        return f"Nombre: {self._nombre} | Trata la enfermedad: {self.enfermedad} | Descripcion: {self._descripcion} | Precio: {self._precio}"

    def __str__(self):
        return f"Nombre: {self._nombre} | Trata la enfermedad: {self.enfermedad} | Descripcion: {self._descripcion} | Precio: {self._precio} | Cantidad: {self._cantidad}"

    @property
    def enfermedad(self):
        return self._enfermedad

    @enfermedad.setter
    def enfermedad(self, value):
        self._enfermedad = value

    @property
    def precio(self):
        return self._precio

    @precio.setter
    def precio(self, value):
        self._precio = value

    @property
    def descripcion(self):
        return self._descripcion

    @descripcion.setter
    def descripcion(self, value):
        self._descripcion = value

    @property
    def nombre(self):
        return self._nombre

    @nombre.setter
    def nombre(self, value):
        self._nombre = value

    @property
    def cantidad(self):
        return self._cantidad

    @cantidad.setter
    def cantidad(self, value):
        self._cantidad = value
